File: code/138/solution.go
package main /*{{{*/

import (
	"fmt"
	"reflect"
	"sort"
)

const (
	e4  = int64(10000)
	e5  = int64(100000)
	e6  = int64(1000000)
	e7  = int64(10000000)
	e8  = e4 * e4
	mod = int64(1e9 + 7)
)

func add(list ...interface{}) int64 { /*{{{*/
	if len(list) == 0 {
		return 0
	}
	ret := int64(0)
	for _, i := range list {
		buf := int64(0)
		switch reflect.TypeOf(i).String() {
		case "int":
			buf = int64(i.(int))
		case "int64":
			buf = i.(int64)
		}
		ret = (ret + buf) % mod
	}
	return ret
} /*}}}*/
func mul(list ...interface{}) int64 { /*{{{*/
	if len(list) == 0 {
		return 0
	}
	ret := int64(1)
	for _, i := range list {
		buf := int64(0)
		switch reflect.TypeOf(i).String() {
		case "int":
			buf = int64(i.(int))
		case "int64":
			buf = i.(int64)
		}
		buf %= mod
		base := ret
		ret = 0
		for buf > 0 {
			if (buf & 1) == 1 {
				ret = add(ret, base)
			}
			base = add(base, base)
			buf >>= 1
		}
	}
	return ret
} /*}}}*/
func qpow(a, b int64) int64 { /*{{{*/
	ret := int64(1)
	for b > 0 {
		if (b & 1) == 1 {
			ret = mul(ret, a)
		}
		a = mul(a, a)
		b >>= 1
	}
	return ret
} /*}}}*/
/*}}}*/

func main() {
	v := []int64{}
	k, L := int64(0), int64(1)
	for i := 0; i < 14; i++ {
		k, L = -9*k-4*L-4, -20*k-9*L-8
		if L > 0 {
			v = append(v, L)
		}
	}

	k, L = 0, -1
	for i := 0; i < 14; i++ {
		k, L = -9*k-4*L+4, -20*k-9*L+8
		if L > 0 {
			v = append(v, L)
		}
	}

	sort.Slice(v, func(i, j int) bool {
		return v[i] < v[j]
	})
	ans := int64(0)
	for i := 0; i < 12; i++ {
		ans += v[i]
	}
	fmt.Println(ans)
}

File: code/138/solution.cpp
#include "../template/bigint.cpp"
#include "../template/china_reminder.cpp"
#include "../template/euler.cpp"
#include "../template/pell.cpp"
#include "../template/prime.cpp"
#include "../template/prime_cnt.cpp"
#include <iostream>/*{{{*/
#include <map>
#include <math.h>
#include <set>
#include <vector>
const long long e2 = 100;
const long long e3 = e2 * 10;
const long long e4 = e3 * 10;
const long long e5 = e4 * 10;
const long long e6 = e4 * e2;
const long long e7 = e6 * 10;
const long long e8 = e7 * 10;
long long mod      = 1e9 + 7;
using namespace std; /*}}}*/

/*add{{{*/
long long base_add(long long a, long long b) {
    a %= mod;
    b %= mod;
    return (a + b) % mod;
}
long long add(long long a, long long b) { return base_add(a, b); }
long long add(initializer_list<long long> args) {
    long long ret = 0;
    for (auto i : args)
        ret = base_add(ret, i);
    return ret;
}
/*}}}*/

/*{{{mul*/
long long base_mul(long long a, long long b) {
    a %= mod;
    b %= mod;
    long long ret = 0;
    while (b) {
        if (b & 1)
            ret = (ret + a) % mod;
        b >>= 1;
        a = (a + a) % mod;
    }
    return ret;
}
long long mul(long long a, long long b) { return base_mul(a, b); }
long long mul(initializer_list<long long> args) {
    long long ret = 1;
    for (auto i : args)
        ret = base_mul(ret, i);
    return ret;
} /*}}}*/

/*{{{qpow*/
long long qpow(long long a, long long b) {
    long long ret = 1;
    while (b) {
        if (b & 1)
            ret = mul({ret, a});
        a = mul({a, a});
        b >>= 1;
    }
    return ret;
} /*}}}*/

int main() {
    vector<long long> v;
    long long k = 0, L = 1;
    for (int i = 0; i < 14; i++) {
        auto kk = -9 * k - 4 * L - 4;
        L       = -20 * k - 9 * L - 8;
        k       = kk;
        cout << k * 2 << ' ' << L << endl;
        if (L > 0)
            v.push_back(L);
    }
    cout << endl << endl;

    k = 0, L = -1;
    for (int i = 0; i < 14; i++) {
        auto kk = -9 * k - 4 * L + 4;
        L       = -20 * k - 9 * L + 8;
        k       = kk;
        cout << k * 2 << ' ' << L << endl;
        if (L > 0)
            v.push_back(L);
    }
    cout << endl << endl;

    sort(v.begin(), v.end());
    long long ans = 0;
    for (int i = 0; i < 12; i++) {
        cout << v[i] << endl;
        ans += v[i];
    }
    cout << ans << endl;
    return 0;
}
